README.md

```diff
@@ -1,6 +1,101 @@
-![Clump](clump.png)Clump
+![Clump](clump.png) Clump
+=========================
+Library to ease batching of requests to remote services
+
 [![Build Status](https://secure.travis-ci.org/fwbrasil/clump.png)](http://travis-ci.org/fwbrasil/clump)
 [![Coverage Status](https://coveralls.io/repos/fwbrasil/clump/badge.png)](https://coveralls.io/r/fwbrasil/clump)
-====
 
-A plagiarization of Twitter's unreleased [Stitch](https://www.youtube.com/watch?v=VVpmMfT8aYw)
\ No newline at end of file
+# Introduction #
+
+## The problem ##
+
+The microservices architecture introduces many new challenges when dealing with complex systems. One of them is the high number of remote procedure calls (RPCs) and the cost associated to them. Among the common techniques applied to amortize this cost, batching of requests has an important role. Instead of paying the price of one call for each interaction, many interactions are batched in only one call.
+
+While batching introduces performance enhancements, it also introduces complexity to the codebase. The common approach is to extract as much information as possible about what needs to be fetched, perform the batched fetch, and then compose the values again to have the final result. These steps need to repeated many times depending on how complex is the final structure.
+
+An example of batching using futures:
+
+```scala
+tracksService.get(ids).flatMap { tracks =>
+	val userIds = tracks.map(_.user)
+	usersService.get(userIds).map {	users =>
+		val userMap = userIds.zip(users).toMap
+		tracks.map { track =>
+			new EnrichedTrack(track, userMap(track.user))
+		}
+	}
+}
+```
+
+This example has only one level of nested resources. In a complex system, it is common to have several levels:
+
+```
+timeline
```

Comment:
I'm not sure we need this example. I would like to keep the introduction as short as possible. Wdyt?